Sidebar menu

UFO-101 opened the pull request:
- Add floating contents menu that stays on the side if the page is wide enough (otherwise it stays at the top).
 - Move '#contents' anchor point to just above main title.
 - Change paper link to the arxiv page, not the pdf directly.

tomtseng commented on sgf-viewer/src/components/subcomponents/Contents.svelte:
Maybe name `TableOfContents.svelte` or something like that to make it clear that this doesn't actually contain the contents of the page but rather a navigation table for the contents of the page

tomtseng commented on sgf-viewer/src/components/subcomponents/Contents.svelte:
```diff
@@ -0,0 +1,182 @@
+<script lang="ts">
+    import { onMount, onDestroy } from "svelte";
+    import { pages } from "../../content";
+    import { fade } from "svelte/transition";
```
```suggestion
    import { onMount, onDestroy } from "svelte";
    import { fade } from "svelte/transition";
    
    import { pages } from "../../content";
```
(just grouping together external-library imports vs. relative file imports)

UFO-101 commented:
Thanks, I've implemented those changes.

tomtseng reviewed (approved):
LGTM, Tony said in our weekly meeting today that he had some aesthetic nits but that they could be addressed in a separate PR

In terms of aesthetics, the margin between the content and the text & the margin between the content and the left side of the window might be a bit narrow? Also something I noticed (which isn't new to this PR) that I'm not sure whether is intentional: seems like the summary section has slightly less left-margin/padding than the subsequent sections

![Screen Shot 2023-05-24 at 13 02 07](https://github.com/AlignmentResearch/KataGoVisualizer/assets/7245938/9a0aa862-fbc3-47a4-9332-3af31bfd5080)

UFO-101 commented:
> margin between the content and the text (depending on the window size) & the margin between the content and the left side of the window might be a bit narrow

Changed the minimum screen size to show the menu floating on the left and made the contents float in the middle of the space on the left side.

Also something I noticed (which isn't new to this PR) that I'm not

>  sure whether is intentional: seems like the summary section has slightly less left-margin/padding than the subsequent sections

Thanks, I hadn't noticed that. Fixed.
